### ISSUE TITLE
Day/Night Summary and Icon Fields

### ISSUE THREAD

**Issue by cloneofghosts** [alexander0042/pirateweather#49]
It would be nice if the summary field could be split into two summary fields with one being for daytime conditions and the other one for nighttime conditions. One of my main complaints about the summaries from DarkSky is that since they kept the summaries breif it would often miss out on large portions of the day depending on the forecast. Some examples

Foggy in the morning. 
Possible light rain in the morning.
Rain and windy overnight.

Those summaries don't provide much information outside of those timeframes so you'd have to look at the hourly forecast anyway to get more information and having it split into day and night helps to solve that problem.

As for the existing summary/icon fields they could either be used to show a summary of the day like they are now or it could show the day information during the day and then swap to the night information at night. The first option is something weather.com does with their API so maybe that would be better?

**Comment by cloneofghosts** [comment]
Looks like this one was passed over. Thoughts about adding something like this when text summaries are implemented?

If the text summaries are going to take a while can you add the no-autoclose label to this issue so it doesn't attempt to auto-close this issue every 90 days.

**Comment by alexander0042** [comment]
Yup, let's keep this one open! Once I get the daily max stuff sorted out, complete text summaries in the Dark Sky style are next

**Comment by cloneofghosts** [comment]
@alexander0042 A way to keep this open without it automatically adding the stale label would be to either assign this issue to yourself or add the keep label to this issue. Either of those will prevent the bot from marking this as stale in the future.

**Comment by alexander0042** [comment]
Perfect! Label is added

**Comment by cloneofghosts** [comment]
@alexander0042 I see it has the enhancement label. The only ignored label is called keep though if you want it to ignore items tagged as enhancement and bug you can easily add it by modifying lines 16 and 17 https://github.com/alexander0042/pirateweather/blob/main/.github/workflows/stale.yml to be

exempt-issue-labels: 'keep, enhancement, bug'
exempt-pr-labels: 'keep, enhancement, bug'

**Comment by alexander0042** [comment]
Eventually I figure things out, keep label makes a lot of sense

**Comment by cloneofghosts** [comment]
Was thinking about this and maybe instead of having a day/night summary/icon fields it might be better to have a section to display a forecast in day/night chunks.

So it would have a day block which displays the forecast from 6am to 6pm for the day and the night block would be from 6pm to 6am. Maybe once there are fewer than x number of hours left the block no longer shows since it doesn't make sense to show a forecast for only a few hours. It could maybe have the same fields as the hourly section does currently.

To me it would make more sense then having a day & night summary field and I know that most other APIs do offer the ability to have a day & night forecast and would be useful for apps which would like to show this information.

This idea came about since I noticed that [Breezy Weather](https://github.com/breezy-weather/breezy-weather) has PW as a weather provider but the daily information becomes limited after 48h since they do not use the extended hourly forecasts.

**Comment by cloneofghosts** [comment]
@alexander0042 Is this something that will still be worked on in the future? Would be a nice addition as most APIs offer a day/night forecast now and could be shown using an include query string.

The only thing would be figuring out the summaries but we can look at that if/when this is added.

**Comment by cloneofghosts** [comment]
@alexander0042 Was thinking about this again and wonder if this could be something added for 2.8? With the summaries being mainly static this could help give more information about the day. As an example if there's morning fog the daily summary would always be Foggy in the morning. So with a day/night forecast you could have the day forecast say Foggy and the night forecast me Mostly Clear.

Obviously that was just an example but the summary would need to be figured out. Like if rain starts in the afternoon and ends in the evening I'm not sure if it should say

Light rain in the afternoon.
Drizzle in the evening.

or try and do something similar like the day summaries

Light rain starting in the afternoon.
Drizzle until night.

Maybe the first one makes more sense? The day summaries should be able to generate something like that already but would need to have a mode for half-day summaries so it would generate for-day and maybe a new for-night?

**Comment by georryan** [comment]
> It would be nice if the summary field could be split into two summary fields with one being for daytime conditions and the other one for nighttime conditions. One of my main complaints about the summaries from DarkSky is that since they kept the summaries breif it would often miss out on large portions of the day depending on the forecast. Some examples
> 
> Foggy in the morning. Possible light rain in the morning. Rain and windy overnight.
> 
> Those summaries don't provide much information outside of those timeframes so you'd have to look at the hourly forecast anyway to get more information and having it split into day and night helps to solve that problem.
> 
> As for the existing summary/icon fields they could either be used to show a summary of the day like they are now or it could show the day information during the day and then swap to the night information at night. The first option is something weather.com does with their API so maybe that would be better?

It would be nice to also have a day or night flag on the hourly fields.  I would like to exclude the daily data but still have info on weather the hour I am looking at is day or night. The icon doesn't always give that info by itself.
